winpr/libwinpr/utils/asn1/asn1.c

```diff
@@ -1024,13 +1024,20 @@ static size_t WinPrAsn1DecReadIntegerLike(WinPrAsn1Decoder* dec, WinPrAsn1_tag e
 		return 0;
 
 	WinPrAsn1_INTEGER val = 0;
-	for (size_t x = 0; x < len; x++)
+	UINT8 v = 0;
+
+	Stream_Read_UINT8(&dec->source, v);
```

Comment:
Missing length check

Reply:
The length check is done 2 lines above

Reply:
@hardening and does not handle the case `len == 0` and unconditionally reads 1 byte here

Reply:
@akallabeth  oh yeah I missed that one, good catch